src/main/java/com/team1816/season/subsystems/Turret.java

```diff
@@ -61,107 +59,92 @@ public class Turret extends Subsystem implements PidProvider {
-        turret.configReverseSoftLimitThreshold(REV_LIMIT, Constants.kCANTimeoutMs); // Reverse = MIN
-        turret.overrideLimitSwitchesEnable(true);
-        turret.overrideSoftLimitsEnable(true);
+        turretMotor.setNeutralMode(NeutralMode.Brake);
+        turretMotor.configForwardSoftLimitEnable(true, Constants.kCANTimeoutMs);
+        turretMotor.configReverseSoftLimitEnable(true, Constants.kCANTimeoutMs);
+        turretMotor.configForwardSoftLimitThreshold(kFwdLimit, Constants.kCANTimeoutMs); // Forward = MAX
+        turretMotor.configReverseSoftLimitThreshold(kRevLimit, Constants.kCANTimeoutMs); // Reverse = MIN
+        turretMotor.overrideLimitSwitchesEnable(true);
+        turretMotor.overrideSoftLimitsEnable(true);
     }
 
     /**
-     * converts 0-360 to 0-TURRET_ENCODER_PPR with zero offset
+     * converts 0-360 to 0-TURRET_ENCODER_PPR
      */
     public static int convertTurretDegreesToTicks(double degrees) {
-        return ((int) (((degrees) / 360.0) * TURRET_PPR));
+        return (int) (degrees / 360.0 * kTurretPPR);
     }
 
     /**
-     * converts 0-TURRET_ENCODER_PPR with zero offset
+     * converts 0-TURRET_ENCODER_PPR
      */
-    public double convertTurretTicksToDegrees(double ticks) {
-        return ticks / TURRET_PPR * 360;
+    public static double convertTurretTicksToDegrees(double ticks) {
+        return ticks / kTurretPPR * 360;
     }
 
     @Override
-    public synchronized void zeroSensors() { // TODO here
-        if (turret instanceof IMotorSensor) {
-            var sensors = (IMotorSensor) turret;
-            int halfAbsPPR = ABS_PPR / 2;
-            int absSensorVal = sensors.getPulseWidthPosition(); // absolute
-            int offset = ZERO_OFFSET - absSensorVal;
-
-            // It is safe to reset quadrature if turret enc reads ~0 (on startup)
-            if (
-                Math.abs(sensors.getQuadraturePosition()) < halfAbsPPR ||
-                (int) ENC_RATIO == 1
-            ) {
-                //second check - don't zero if abs enc not in viable range
-                if (absSensorVal > -1 && absSensorVal < ABS_PPR) {
-                    sensors.setQuadraturePosition(offset);
-                    System.out.println("Zeroing turret limits! Offset: " + offset);
-                } else {
-                    DriverStation.reportError(
-                        "ABSOLUTE ENCODER INVALID RANGE - not zeroing",
-                        false
-                    );
-                }
-            } else {
-                System.out.println("UNSAFE - NOT ZEROING TURRET QUADRATURE");
-            }
+    public synchronized void zeroSensors() {
+        zeroSensors(false);
+    }
+
+    public synchronized void zeroSensors(boolean resetEncPos) {
+        desiredPos = 0;
+        followingPos = 0;
+        lostEncPos = false;
+
+        if (resetEncPos) {
+            turretMotor.setSelectedSensorPosition(
```

Comment:
Unless I misunderstood new logic, this will break the turret - this sets the selectedSensorPosition (relative encoder position) to 0. What the previous logic did was set it to the absolute encoder position. It rehomed the the quadrature (relative) encoder to the source of truth absolute (pulse width) encoder.

The absolute encoder was never being zeroed.